jira_automation/src/main/java/io/dockstore/jira/MilestoneResolver.java

```diff
@@ -0,0 +1,181 @@
+package io.dockstore.jira;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHMilestone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Attempts to resolve JIRA DOCK ticket fix version mismatches with the corresponding GitHub milestone.
+ * Handles these cases:
+ *
+ * <ol>
+ *     <li>There is a JIRA fix version but no GitHub milestone -- sets the GitHub milestone to the JIRA fix version</li>
+ *     <li>There is a GitHub milestone but no JIRA fix version -- sets the JIRA fix version to the GitHub milestone</li>
+ * </ol>
+ *
+ * <p>It does not handle the case of the JIRA fix version not matching the GitHub milestone. It does
+ * print out a message saying the mismatch needs to be resolved manually. To resolve this automatically,
```

Comment:
Or vice versa. 
I like the time idea in the comments but probably not worth it at this juncture.

Reply:
Even though at backlog grooming we go through it on the JIRA side, I think there are still cases where sometimes set it on the GitHub side first. I thing we should just leave the PR as is for now in this area. And if it becomes frequent enough an issue, we can revisit per the comments below.

FWIW, I just update it manually, and it usually only takes a couple of minutes; this may shave off half the time as is.